PKHeX.Core.AutoMod/AutoMod/RegenTemplate.cs

```diff
@@ -92,13 +92,13 @@ private void LoadExtraInstructions(List<string> lines)
             }
         }
 
-        private static int[] SanitizeEVs(int[] evs, bool gb)
+        private static int[] SanitizeEVs(int[] evs, int gen)
         {
             var copy = (int[])evs.Clone();
             for (int i = 0; i < evs.Length; i++)
             {
-                if (copy[i] > (gb ? 65535 : 252))
-                    copy[i] = gb ? 65535 : 252;
+                if (copy[i] > (gen >=6 ? 252 : gen >= 3 ? 255 : 65535))
```

Comment:
Compute the maxEV outside of the loop rather than recalculating on each compare and set (1x vs ~6x)